### DISCUSSION TITLE
Improve photo plugin

### DISCUSSION THREAD

**Discussion by ghost** [osmandapp/OsmAnd#14705] [General]
OsmAnd is an excellent application, but the plugin for taking geotagged photos is terrible.

When I take photos with OsmAnd, the photos are in the application but they are not found in Android. I don't really understand the lack of intuitiveness of this feature. Why are photos not appearing in the Android gallery? Even applications like Telegram or Signal ask the user for permission to access the storage and create a folder in the gallery.

It really is because this feature is very useful. But it is not possible to recover the photos.

You really have to do something to choose the destination folder. To display the folder. For the photos to be recoverable.

I don't see the connection to privacy. Let user choose permissions and integrate OsmAnd with Android gallery.

Otherwise, this feature is useless.

![Screenshot_20220629-120820](https://user-images.githubusercontent.com/81697984/176411846-f544a5b9-8139-4502-b0c4-0fe6a1f8a0fd.png)
![Screenshot_20220628-171914](https://user-images.githubusercontent.com/81697984/176411858-5dbecabb-ae76-4548-92c7-07caa4e255d1.png)

**Comment by ghost** [comment]
I contacted google. Since Android 11 the `android/data` folder is no longer accessible by the user.

As long as you don't ask permission to access the gallery, this plugin will remain useless because the photos are unrecoverable.

**Reply by tommycrock** [reply]
I think there are plans to do this but it will take some significant refactoring https://github.com/osmandapp/OsmAnd/discussions/13538#discussioncomment-2849344
Also, since some people use it for photo mapping they don't necessarily want those photos mixed in with their others. The problem has really been caused by Google's change of policy making it awkward to use osmand outside of the Android structure or with all file access.

**Reply by tommycrock** [reply]
I know it's not ideal but one option for now is share them. In the menu, My Places, select the A/V notes tab. There's a share icon at the bottom, hit that and you can select all and share to a file manager, etc. You can also share them individually.

Alternatively you can move the storage to multiuser storage, which I understand is USB accessible, or to within the media folder, which gives you full read/ write access but some people have performance issues.

**Reply by ghost** [reply]
I can't find this menu, do you have screenshots? Thanks

**Reply by ghost** [reply]
I found, thanks for the solution, I started to despair

**Reply by tommycrock** [reply]
![Screenshot_20220701-144032_OsmAnd+](https://user-images.githubusercontent.com/5749884/176964920-47a691a4-a84c-4374-8881-7385cbc0fa0d.jpg)
![Screenshot_20220701-144040_OsmAnd+](https://user-images.githubusercontent.com/5749884/176964950-b118d50f-88e5-4bb7-a0ef-653d7f61d92e.jpg)
![Screenshot_20220701-144049_OsmAnd+](https://user-images.githubusercontent.com/5749884/176964964-33e4dd95-c3dc-4d5c-aeb3-824f125d5167.jpg)

**Reply by tommycrock** [reply]
For what it's worth, nowadays I use Vespucci for photomapping. It allows me to choose OpenCamera as the app (despite Google removing that as an API option). Although it doesn't put the photos in the DCIM folder it installs itself to the download folder and puts them in there, which suits me.
